src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -513,6 +514,17 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 }
             }
 
+            if (token.IsKind(SyntaxKind.AsteriskToken) &&
+                next.IsKind(SyntaxKind.AsteriskToken) || token.Parent is PointerTypeSyntax)
```

Comment:
this whole expression is too confusing.  Also, it seems unnecessary.  Why not simply check if you have an asterisk and the parent is a pointertype.  that seems like it should be sufficient.

Reply:
@CyrusNajmabadi It's necessary for things like `int **p`

That simply says:

- If current and next are both asterisk, then no space.
- If current is asterisk and parent is PointerTypeSyntax (the asterisk in this case is the last one), then no space also.

Reply:
i don't understand why the `int **`  is special.  For both of those asterisks, the parent is a PointerTYpeSyntax, so there should be 0 space between it and the previous token.